src/Orleans/Utils/IAsymmetricComparer.cs

```diff
@@ -0,0 +1,16 @@
+﻿
+namespace Orleans.Runtime
+{
+    public interface IAsymmetricComparer<in T1, in T2>
```

Comment:
How about you just make `public interface `CacheDataComparer<in TCachedMessage>` not extend any `IAsymmetricComparer` interface, but just define the interface methods directly?
`ICacheDataComparer` lives in the `Orleans.Providers.Streams.Common` namespace, so its less of a concern.

IAsymmetricComparer<TCachedMessage, StreamSequenceToken>, IAsymmetricComparer<TCachedMessage, IStreamIdentity>